### PR TITLE
Generator Robustness

### PR DESCRIPTION
**Pull request by grayside**
This is a huge PR that tackles a number of race condition, compatibility, and performance issues with grunt-drupal-tasks 0.9 and other generators.
- Move and slightly adjust the initialization of the project repo to use code homed in generator-gadget, rather than using a time-consuming npm install of gadget to get it's separately maintained examples/ directory. Now GDT can repurpose that directory to be less a template and more an example.
- Handle the first pass at computing library versions and initializing the codebase in the "configuring" stage instead of falling back to default. It is unlikely this will help across generators much if I understand Yeoman's compose architecture, but it would be delightful to learn otherwise.
- Use the in-memory filesystem to initialize the project, specifically `this.fs.copy()`. This fixes the current problem of treating the initial skeleton of files as templates, as well as placing them in memory so other generators can effectively interact with the files without needing to go to Gadget's source templates or hope Gadget commits changes before the other generator needs access.

This set of changes had several secondary impacts, including:
- Deprecate our use of jsonfile
- Need to populate the package.json with a grunt-drupal-tasks entry without relying on `npm install`
- Move all Drupal/Drush centric templates into `app/templates/drupal` for cleaner namespacing.
- Drop `app/gdt.js` in favor of adding an npmVersion utility function to `lib/util.js`
- Switch to use `this.destinationRoot()` and `this.sourceRoot()` for best practice in reference the "root directory" of the respective locations.

Deprecates and Closes #33 

@tekante @EvanLovely @arithmetric @dsasser

### CODE REVIEW

**Comment by EvanLovely** [comment]
I'm seeing this work with node v4.2.2 - nice job!

**Comment by grayside** [comment]
Rereading http://yeoman.io/authoring/running-context.html, it seems like the separation of initializing the GDT scaffolding in the configuration stage will indeed avoid race conditions with other generators.

**Comment by tekante** [comment]
Looks to work for me as well, node v4.2.2, npm v2.14.7

File: lib/util.js [hunk 1]
```diff
@@ -13,3 +14,17 @@ module.exports.tokens = function(options) {
...
+  if (useMaster) {
+    return 'git+https:' + relativeRepoUrl + '#master';
+  }
+  else {
+    var version = spawn.spawnCommandSync(
+      'npm',
+      ['show', name, 'version'],
+      {stdio: 'pipe'}
+    ).stdout.toString('utf8');
+    return '^' + _.trim(version);
```
Comment (arithmetric) [inline lib/util.js]: Is this an intentional change from `~` currently used in the template package.json?

**Comment by arithmetric** [comment]
@grayside Nice work. This looks like a good approach to simplify the process. I noticed a few issues when testing the output generated:

After setting up a Drupal 8 and OpenAtrium site, I saw this error when running grunt:

```
Running "symlink:libraries" (symlink) task
>> Created 0 symbolic links.

Running "symlink:modules" (symlink) task
Warning: Arguments to path.join must be strings Use --force to continue.

Aborted due to warnings.
```

Also, when creating a Drupal 8 site, the `test/features/example.feature` test scenario generated is the Drupal 7 version.

**Comment by arithmetric** [comment]
If you merge the branch from #36 into this branch, the tests should work through the generator and then fail during grunt with this error ^

File: lib/util.js (hunk 1 above)
Comment (grayside) [inline lib/util.js]: This is a memory error on my part. Will fix.

**Comment by grayside** [comment]
I believe I understand the d7 test issue, will troubleshoot all 3 points.

**Comment by grayside** [comment]
Got these issues covered and merged #36, so I anticipate the tests passing. Note that OpenAtrium will get a behat.yml, but does not currently have any supported default tests in this structure. Were the d7 ones compatible? If so we can add a duplicate of them pretty easily, otherwise we should handle Atrium tests in follow-up.
- [x] Use ~ instead of ^ to qualify gdt version.
- [x] Drupal 8 gets Drupal 7 version of the example Behat tests.
- [x] grunt scaffold fails on symlink.
- [ ] adding or deferring Atrium Behat example tests.

**Comment by grayside** [comment]
Tests are failing, I have some work-in-progress on that, will resume tomorrow.

**Comment by grayside** [comment]
Per discussion, we'll circle back on Atrium tests and this is good to go.

**Pull request merged**